yrs/src/block.rs

```diff
@@ -15,13 +17,22 @@ const BLOCK_ITEM_ANY_REF_NUMBER: u8 = 8;
 const BLOCK_ITEM_DOC_REF_NUMBER: u8 = 9;
 const BLOCK_SKIP_REF_NUMBER: u8 = 10;
 
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, PartialOrd, PartialEq, Ord, Eq, Hash)]
```

Comment:
I want to be a bit careful about whether a Hash function is applied on `TypePtr` / `BlockPtr` / `ID`. 

Is the `Hash` needed to do comparisons? Can we optimize this using a comparison `id.client == other.client && id.clock === other.clock`. Possibly we can improve that by comparing the 12 bytes directly.

We could compute ordering as `id < other ⇔ id.client < other.client  ||  (id.client == other.client && id.clock < other.clock)` (which is a common comparison method used in Yjs). Then we could reuse ordering in the integration process.

Reply:
Had to do that because transaction.changes is a map which uses `TypePtr` as a key. I can probably use dedicated Hasher instead.

Reply:
Re operators:
- I removed `PartialOrd` and `Ord` as we usually don't need them.
- `PartialEq` and `Eq` are already using structural comparison and optimizations like memcmp will be usefull, I'd leave them for later.
- For `Hash` I've implemented `XorHasher` (which simply builds hash as series of xor transforms), but again: I'd leave these sorts of optimizations till the end, when we'll have some benchmarks and not apply extra logic in blind.